cedar-lean/Cedar/Thm/Data/List.lean

```diff
@@ -1203,4 +1498,65 @@ theorem filterMap_congr {f g : α → Option β} : ∀ {l : List α},
     let ⟨h₁, h₂⟩ := forall_mem_cons.1 h
     rw [filterMap, filterMap, h₁, filterMap_congr h₂]
 
+theorem filterMap_empty_iff_all_none {f : α → Option β} {xs : List α} :
+  xs.filterMap f = [] ↔ ∀ x ∈ xs, f x = none
+:= by
+  constructor
+  case mp =>
+    induction xs
+    case nil => simp
+    case cons hd tl ih =>
+      intro h₁ a h₂
+      simp only [List.filterMap_cons] at h₁
+      split at h₁ <;> try simp only at h₁
+      case h_1 h₃ =>
+        rcases (List.mem_cons.mp h₂) with h₄ | h₄
+        case inl => subst h₄ ; assumption
+        case inr => apply ih h₁ a ; assumption
+  case mpr =>
+    intro h₁
+    induction xs
+    case nil => simp
+    case cons hd tl ih =>
+      simp only [List.filterMap_cons]
+      split
+      case h_1 =>
+        apply ih ; clear ih
+        intro a h₂
+        apply h₁ a
+        exact List.mem_cons_of_mem hd h₂
+      case h_2 b h₂ =>
+        exfalso
+        specialize h₁ hd
+        simp only [mem_cons, true_or, forall_const] at h₁
+        simp [h₁] at h₂
+
+theorem filterMap_nonempty_iff_exists {f : α → Option β} {xs : List α} :
```

Comment:
Nit: I feel like `*_exists_some` would be a little more pretty & symmetric with `*_all_none`

Reply:
Updated.